lock-api/src/main/java/com/palantir/lock/watch/ClientLockWatchEventLog.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import com.palantir.lock.LockDescriptor;
+
+public interface ClientLockWatchEventLog {
+    TransactionsLockWatchEvents getEventsForTransactions(
+            Map<Long, IdentifiedVersion> timestampToVersion,
+            Optional<IdentifiedVersion> version);
+    Optional<Set<LockDescriptor>> getEventsBetweenVersions(
+            IdentifiedVersion startVersion,
+            IdentifiedVersion endVersion);
```

Comment:
why is this Optional?

Reply:
also, why does this not return a TransactionsLockWatchEvents?

Reply:
I could equally return that and have the above (i.e. the cache) turn that into a commit update; this approach just makes the cache not worry about events themselves.